src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -72,7 +72,7 @@ public static function createConnection($dsn, array $options = array())
         }
         $params = preg_replace_callback('#^redis://(?:([^@]*)@)?#', function ($m) use (&$auth) {
             if (isset($m[1])) {
-                $auth = $m[1];
+                $auth = preg_replace('/^(\w+:)?(\w+)$/', '$2', $m[1]);
```

Comment:
instead of calling preg_replace in preg_replace_callback, I propose to change the previous regexp for:
`'#^redis://(?:(?:[^:@]*+:)?([^@]*+)@)?#'`
it should the same, can you confirm?